### PR TITLE
Validate domains cronjob

### PR DESCRIPTION
**Pull request by ph448**
This PR adds two tasks, one to validate DomainNames and one to delete the expired ones.
Note that these are not actually plumbed in, only added so that they can be reviewed.
As there's a lot of commits I recommend a squash merge.

### CODE REVIEW

**Comment by abrahammartin** [comment]
Does this supersedes #200 ?  Or should I merge #200 and then this one? As they seem to have same commits.

**Comment by ph448** [comment]
oh, yes, sorry, #200 first then this

**Comment by abrahammartin** [comment]
You need to add the cronjob to the settings file, see: https://github.com/uisautomation/mws-panel/blob/20795b01473ebcde182f851e5a6b6a3100d8810b/mws/mws/production_settings.py#L45

File: mws/sitesmanagement/cronjobs.py [hunk 1]
```diff
@@ -309,3 +310,67 @@ def reject_or_accepted_old_domain_names_requests():
...
+    Iterate over DomainName objects and set them to:
+     - global if they are visible to (currently) Google's nameservers
+     - private if they are onl`y available to the Cambridge nameservers
+     - deleted if they are visible to none of the above.
+    '''
+    active_states = ['accepted', 'private', 'global', 'deleted']
+
+    for domainname in DomainName.objects.filter(status__in=active_states):
+            status = domainname.validate()
+            if status != domainname.status:
```
Comment (abrahammartin) [inline mws/sitesmanagement/cronjobs.py]: this is done in validate() already, you should call it with update=true
https://github.com/uisautomation/mws-panel/blob/20795b01473ebcde182f851e5a6b6a3100d8810b/mws/sitesmanagement/models.py#L733

File: mws/sitesmanagement/cronjobs.py [hunk 2]
```diff
@@ -309,3 +310,67 @@ def reject_or_accepted_old_domain_names_requests():
...
+@shared_task(base=ScheduledTaskWithFailure)
+def expire_domains():
+    '''
+    Periodically send messages to domain administrators about deleted domains and
+    delete them after grace_days
+    '''
+    grace_days = settings.MWS_DOMAIN_NAME_GRACE_DAYS
+    notify_days = list(range(1, grace_days, grace_days//3+1))
+    support_email = settings.EMAIL_MWS3_SUPPORT
+    for domainname in DomainName.objects.filter(status='deleted').annotate(expired=(now() - F('updated_at'))):
```
Comment (abrahammartin) [inline mws/sitesmanagement/cronjobs.py]: 'external' and 'special' will never get to the status 'deleted' as we are not calling validate() on them. Line 325 only calls validate for domains in ['accepted', 'private', 'global', 'deleted'] states.

Comment (abrahammartin) [reply mws/sitesmanagement/cronjobs.py]: Also we are calling validate on line 325 for domain names with status 'deleted'. That means that the update_at will get updated every time we call it, which will cause the expired annotation not to represent reality.

Comment (ph448) [reply mws/sitesmanagement/cronjobs.py]: added 'external' and 'special' to active_states list. validate only saves a model if its status changes, so the annotation should work correctly.

Comment (abrahammartin) [reply mws/sitesmanagement/cronjobs.py]: oh yes! I missed that

**Pull request merged**